deepr/io/parquet.py

```diff
@@ -39,46 +38,76 @@ class ParquetDataset:
     """
 
     def __init__(
-        self, path_or_paths: Union[str, Path, List[Union[str, Path]]], filesystem: FileSystem = None, **kwargs
+        self,
+        path_or_paths: Union[str, Path, List[Union[str, Path]]],
+        filesystem: FileSystem = None,
+        metadata: pq.FileMetaData = None,
+        schema: pyarrow.Schema = None,
+        split_row_groups: bool = False,
+        validate_schema: bool = True,
+        filters: List = None,
+        metadata_nthreads: int = 1,
+        memory_map: bool = False,
     ):
         self.path_or_paths = list(map(str, path_or_paths)) if isinstance(path_or_paths, list) else str(path_or_paths)
         self.filesystem = filesystem
-        self._kwargs = kwargs
-
-        # Lazy loaded
-        self._pq_dataset = None
+        self.schema = schema
+        self.metadata = metadata
+        self.split_row_groups = split_row_groups
+        self.validate_schema = validate_schema
+        self.filters = filters
+        self.metadata_nthreads = metadata_nthreads
+        self.memory_map = memory_map
```

Comment:
this list of arguments corresponds to pyarrow 0.14.1 -- newer versions have more arguments